### PR TITLE
User schema

### PR DESCRIPTION
**Pull request by mngweb**

### CODE REVIEW

File: backend/src/models/User.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+userSchema.plugin(uniqueValidator, { message: '{PATH} {VALUE} is already taken' });
+
+const User = mongoose.model('User', userSchema);
+
+// const exampleUser = new User({
+//   username: 'AidKitMan',
+//   email: 'aidkitman@gmail.com',
+//   password: '5!E@c#r$e%1',
+//   profileRef: new mongoose.Types.ObjectId(),
+// });
```
Comment (Nefariusek) [inline backend/src/models/User.js:63]: Remove commented out text

File: backend/package.json [hunk 2]
```diff
@@ -17,7 +17,8 @@
     "helmet": "^4.6.0",
     "http-status-codes": "^2.1.4",
     "joi": "^17.4.2",
-    "mongoose": "^6.0.6"
+    "mongoose": "^6.0.6",
+    "mongoose-unique-validator": "^3.0.0"
```
Comment (Nefariusek) [inline backend/package.json:21]: remove package

**Review by Nefariusek — CHANGES_REQUESTED** [review]

File: backend/src/models/User.js (hunk 1 above)
Comment (mngweb) [reply backend/src/models/User.js]: Done

File: backend/package.json (hunk 2 above)
Comment (mngweb) [reply backend/package.json:21]: After consultation, left as additional layer of validation

**Pull request merged**